doc/rst/language/spec/modules.rst

```diff
@@ -917,6 +811,213 @@ disambiguate the symbols in this case.
+          writeln(B.C.cSymbol);
+          writeln(B.cSymbol);
+        }
+      }
+
+   In this case, C will be visible to A as though it was a submodule of B, and
+   its symbols can also be treated as though they were defined within B.  This
+   means that A can contain mentions like ``B.C.cSymbol`` if cSymbol was a
+   symbol defined in C, regardless of if C was actually a submodule of B.
+
+   This also means that A can contain mentions like ``B.cSymbol`` which would
+   access C's cSymbol, assuming these symbols were not shadowed by symbols with
+   the same name in B.
+
+   .. BLOCK-test-chapeloutput
+
+      0
+      0
+
+   *Example (use-reexport2.chpl)*.
+
+   However, if the public use of C also disabled accesses to the module name
+   using the ``as`` keyword, e.g.
+
+   .. code-block:: chapel
+
+      module C {
+        var cSymbol: int;
+      }
+
+      module B {
+        public use C as _;
+      }
+
+      module A {
+        proc main() {
+          use B;
+          // writeln(B.C.cSymbol); // Would not work
+          writeln(B.cSymbol);
+        }
+      }
+
+   Then A could only contain mentions like ``B.cSymbol``, it could not access
+   ``cSymbol`` using ``B.C.cSymbol``.  This is because C is not present as a
+   public name in B's scope.
+
+   .. BLOCK-test-chapeloutput
+
+      0
+
+Conversely, if B's use of C was ``private`` then A would not be able to see C's
+symbols at all due to that ``use``.
+
+The situation for ``import`` is similar.  Because import statements only
+enable either qualified or unqualified access to a symbol, it more closely
+resembles the second example instead of the first.
+
+   *Example (import-reexport1.chpl)*.
+
+   Say we have a module A that imports module B, and module B contains a public
+   import of module C:
+
+   .. code-block:: chapel
+
+      module C {
+        var cSymbol: int;
+      }
+
+      module B {
+        public import C;
+      }
+
+      module A {
+        proc main() {
+          import B;
+          writeln(B.C.cSymbol);
+        }
+      }
+
+   In this case, C will be visible to A as though it was a submodule of B.  This
+   means that A can contain mentions like ``B.C.cSymbol`` if cSymbol was a
+   symbol defined in C, regardless of if C was actually a submodule of B.
+
+   .. BLOCK-test-chapeloutput
+
+      0
+
+   *Example (import-reexport2.chpl)*.
+
+   Alternatively, if module B contains a public import of some public symbols
+   defined in module C, then those symbols will be visible to A as though they
+   were defined in module B, unless they are shadowed by symbols of the same
+   name in B.
+
+   .. code-block:: chapel
+
+      module C {
+        var cSymbol: int;
+      }
+
+      module B {
+        public import C.cSymbol;
+      }
+
+      module A {
+        proc main() {
+          import B;
+          writeln(B.cSymbol);
+        }
+      }
+
+   This means that A could contain mentions like ``B.cSymbol`` and it would
+   access C's cSymbol.
```

Comment:
A does contain such a mention, so maybe it should say "Here, A's mention ``B.cSymbol`` accesses `cSymbol` from `C`.